### DISCUSSION TITLE
@default in the schema generate number | null type

### DISCUSSION THREAD

**Discussion by Spoutnik97** [prisma/prisma#24329] [Q&A]
### Bug description

My schema.prisma contains a @default key for one column, but when I get my entity with `findUnique` the returned type is number | null, whereas with the @default decorator the value in the database is always defined

### How to reproduce

1. Use the following CompanySetting schema
2. Run the migrate command
3. Do a query to the model with the `findUnique method`

```
await this.prisma.company.findUnique({
      where: {
        id: id,
      },
      include: {
        companySettings: true,
      },
    });
```

### Prisma information

```prisma

model Company {
  createdAt          DateTime             @default(now())
  updatedAt          DateTime             @updatedAt @map("updated_at")
  id                 String               @id @default(uuid()) @db.Uuid
  companySettingsId  String               @unique @map("company_settings_id") @db.Uuid
  companySettings    CompanySettings      @relation(fields: [companySettingsId], references: [id], onDelete: NoAction, onUpdate: NoAction)

}

model CompanySettings {
  createdAt          DateTime             @default(now())
  updatedAt          DateTime             @updatedAt @map("updated_at")
  salesMarginRatio                 Float?        @map("sales_margin_ratio")
  processMarginRatio               Float?        @default(30) @map("process_margin_ratio")
}
```

```ts
prisma migrate dev
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v18.15.0

### Prisma Version

```
prisma                  : 5.14.0
@prisma/client          : 5.14.0
Computed binaryTarget   : darwin
Operating System        : darwin
Architecture            : x64
Node.js                 : v18.15.0
Query Engine (Node-API) : libquery-engine e9771e62de70f79a5e1c604a2d7c8e2a0a874b48 (at node_modules/.pnpm/@prisma+engines@5.14.0/node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Schema Engine           : schema-engine-cli e9771e62de70f79a5e1c604a2d7c8e2a0a874b48 (at node_modules/.pnpm/@prisma+engines@5.14.0/node_modules/@prisma/engines/schema-engine-darwin)
Schema Wasm             : @prisma/prisma-schema-wasm 5.14.0-25.e9771e62de70f79a5e1c604a2d7c8e2a0a874b48
Default Engines Hash    : e9771e62de70f79a5e1c604a2d7c8e2a0a874b48
Studio                  : 0.500.0
```

**Comment by ludralph** [comment]
Hi @Spoutnik97 👋 

Thank you for raising this question.

Even though the `@default` value ensures that the column in the database will always have a value, the TypeScript type definitions generated by Prisma will still reflect the possibility of null. This is because the `findUnique` method can return null if no record is found that matches the query criteria.

**Reply by Spoutnik97** [reply]
findUnique could reply null if the entity does not exist, but if it exists, then its properties (with @default) cannot be null.

Am I right ?

**Reply by ludralph** [reply]
Yes, you are correct. The `findUnique` method can return null if no record matches the query criteria. However, if a record does exist, the properties that have a `@default` value defined in your Prisma schema will not be null.

**Reply by Spoutnik97** [reply]
Yes, but typescript tells me that the properties that have a @default value defined in my Prisma schema COULD BE null.

Is it a bug ? Or a "normal" behavior ?

**Reply by Spoutnik97** [reply]
@ludralph do I need to remove the optional ? For each columns that have a @default value ?

**Reply by ludralph** [reply]
Hi @Spoutnik97 

You don't necessarily need to remove the optional `?` for columns that have a `@default` value. The `?` in Prisma schema indicates that a field is optional, meaning it can be `null` in the database. The @default attribute, on the other hand, specifies a default value for the field when no value is provided during creation.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
